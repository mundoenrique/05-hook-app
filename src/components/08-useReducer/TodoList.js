import TodoItem from './TodoItem';

export default function TodoList({ todos, handleDelete, handleToggle }) {
	return (
		<ul className="list-group list-group-flush">
			{todos.map((todo, index) => (
				<TodoItem
					key={todo.id}
					todo={todo}
					index={index}
					handleDelete={handleDelete}
					handleToggle={handleToggle}
				/>
			))}
		</ul>
	);
}
