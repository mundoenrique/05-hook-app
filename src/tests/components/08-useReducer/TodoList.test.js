import { shallow } from 'enzyme';
import TodoList from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas sobre <TodoList />', () => {
	const handleToggle = jest.fn();
	const handleDelete = jest.fn();

	const wrapper = shallow(
		<TodoList
			todos={demoTodos}
			handleDelete={handleDelete}
			handleToggle={handleToggle}
		/>
	);

	test('Debe renderizar el componente <TodoList />', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe tener dos  <TodoItem />', () => {
		expect(wrapper.find('TodoItem').length).toBe(demoTodos.length);

		expect(wrapper.find('TodoItem').at(0).prop('handleDelete')).toEqual(
			expect.any(Function)
		);

		expect(wrapper.find('TodoItem').at(0).prop('handleToggle')).toEqual(
			expect.any(Function)
		);
	});
});
