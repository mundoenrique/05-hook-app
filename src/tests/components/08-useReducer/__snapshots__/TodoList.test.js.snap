// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <TodoList /> Debe renderizar el componente <TodoList /> 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Jest",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
